app/stac_api/utils.py

```diff
@@ -388,3 +389,42 @@ def is_valid_b64(value):
         logger.debug('Invalid b64 value %s: %s', value, err)
         return False
     return True
+
+
+def get_dynamic_max_age_value(update_interval):
+    '''Get the max_age value for dynamic cache based on `update_interval` DB field
+
+    -       update_interval < 0  then use default cache settings
+    -  0 <= update_interval < 10 then no cache
+    - 10 <= update_interval then log10(update_interval)*log9(update_interval)
+    Args:
+        update_interval: int
+            Value of the DB field `update_interval`
+
+    Returns:
+        int - Cache control max_age, or -1 to use default cache settings
+    '''
+    threshold_no_cache = 10
+    max_threshold = 60 * 60  # 1h
```

Comment:
Would it make sense to have a global setting or an ENV var for the max cache duration?

Reply:
I don't think, for a bigger value we need to re-think the logic with the logarithm value anyway and will very likely do some changes in the code anyway.

Reply:
Can live with the way it currently is. I just thought, that if we changed this value here, probably other places will be forgotten to be changed, e.g. the hardcoded value of 1h in the error message in line 416. So having it as a global setting, we would only need to change it in one place.